lib/Mojo/IOLoop/ReadWriteProcess.pm

```diff
@@ -489,6 +491,9 @@ sub start {
   $self->emit('start');
   $self->session->register($self->pid() => $self);
 
+  $self->session->emit_from_sigchld($old_emit_from_sigchld);
```

Comment:
For better exception-safety this could be reset using a scope guard.

Reply:
And by that I mean using simply using https://docs.mojolicious.org/Mojo/Util#scope_guard.

Reply:
Wow that's really cool. Thanks wasn't aware of it.

Reply:
With this, we also avoid the problem of `_open()` die behavior. Added a test for it.